### PR TITLE
CLI `compile` docs

### PR DESCRIPTION
**Pull request by dwhswenson**
Docs for features in CLI 0.3.0 were missing from the main documentation. (I forgot open this PR, and honestly, I'm impressed with some of the nice stuff I put in here -- local docs generation catches things in my local environment that haven't been published yet: and even that looks good without any real effort on this.)

This also includes some ... easter egg? docs for the real "wizards" out there.

### CODE REVIEW

**Comment by dwhswenson** [comment]
This is ready for review and comment. Without review, I will merge this after at least 24 hours, no sooner than Sat 28 Sep 03:00 GMT (Fri 27 Sep 22:00 my local).

I'd suggest trying this out on you own machine. We'll only see failures *after* merge, since we're not currently testing docs builds in PRs (thing to change).

File: docs/cli/wizard/index.rst [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+* ``!!quit``: Immediately exit the Wizard (no saving of results).
+* ``!restart``: Restart the current step in the Wizard. Useful if you
+  mistype in a prompt, so that you can only restart this step of the wizard.
+
+
+Name limitations in the Wizard
+------------------------------
+
+When using the Wizard, we recommend that names of objects not begin with
+``?``, ``!``, and not be a numeral. This is because the Wizard frequently
```
Comment (sroet) [inline docs/cli/wizard/index.rst:103]: Could this not be checked in the answer sanitation of the wizard? (just to be **extra** friendly) (not needed for this PR, but might be nice to spin into its own issue)

File: docs/conf.py [hunk 2]
```diff
@@ -32,13 +43,23 @@
 sys.path.insert(0,os.path.abspath('../openpathsampling/'))
 #sys.path.append(os.path.abspath('_themes'))
 
+# -- Preparing the CLI docs -----------------------------------------------
+orig_cwd = os.getcwd()
+try:
+    cli_input_dir = os.path.abspath("./cli/compile/input/")
```
Comment (sroet) [inline docs/conf.py:49]: shouldn't this directory be relative to the current file instead of the cwd?

**Review by sroet — APPROVED** [review]
Couple comments/questions. Feel free to ignore, any/all

Most critical one was the `python -m pip` one. LGTM otherwise

File: docs/cli/compile/input/categories.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    by additional strategies.
+
+    Move schemes can be made under the top-level keyword ``schemes``.
+
+
+strategy:
+  header: Move Strategies
+
+shooting-point-selector:
+  header: Shooting Point Selectors
```
[the remark this answers is not in the document]
Comment (dwhswenson) [reply docs/cli/compile/input/categories.yml]: Done in [5151c1a](https://github.com/openpathsampling/openpathsampling/pull/1159/commits/5151c1ab072373e845f911d3ecac22e88e8b013f)

File: docs/conf.py (hunk 2 above)
Comment (dwhswenson) [reply docs/conf.py]: Done in [b9e2818](https://github.com/openpathsampling/openpathsampling/pull/1159/commits/b9e281856d3dcd9758bffcf902fcecc44480f0d7)

File: docs/cli/compile/file_format.rst [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+needed define it, and the values you assign to those keywords depend the
+simulation you intend to run.
+
+
+Example: Basics and evaluated expressions
+-----------------------------------------
+
+Let's use an OpenMM engine as a simple example. Since this is an engine,
+we'll put it under the ``engines`` top-level keyword.  From the
+:ref:`documentation on the OpenMM engine <engine--openmm>`, we know that it
```
[the remark this answers is not in the document]
Comment (dwhswenson) [reply docs/cli/compile/file_format.rst:40]: It does locally; I think it (like all of this) needs the CLI to be installed to work, though.

File: docs/cli/compile/file_format.rst [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    engines:
+      # note that this is a list of engines, even though we only have one!
+      # this is why there is a '-' before the mapping
+      - type: openmm  # must be exactly openmm to be identified correctly
+        name: my_engine
+        topology: topology.pdb       # note that filenames should be relative
+        system: system.xml           # to the directory from which you run
+        integrator: integrator.xml   # the compile command
+        n_steps_per_frame: 1.0 / 0.002  # we allow math here!
+        n_frames_max: 10000
```
[the remark this answers is not in the document]
Comment (dwhswenson) [reply docs/cli/compile/file_format.rst:54]: Probably? But I'm going to be lazy and skip that documentation here.

File: docs/cli/wizard/index.rst (hunk 1 above)
Comment (dwhswenson) [reply docs/cli/wizard/index.rst:103]: Eh, going to leave this as is...  I'm not sure how much the Wizard will ever matter. It came out of a weird time late in pandemic lockdowns. I still find a few things about it hilarious, but I wouldn't honestly recommend it as an interface right now.

This documentation shouldn't even show up when a user browses the docs; they would need to know to look at the right URL.

**Pull request merged**